### PR TITLE
FS-974: Create timeout test for Connector

### PR DESCRIPTION
**Pull request by PerrettJ4**

### CODE REVIEW

**Review by yangwong9664 — CHANGES_REQUESTED** [review]

**Review by yangwong9664 — DISMISSED** [review]

File: src/test/resources/application.conf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include "frontend.conf"
+appName = "test-application"
+
+play.modules.enabled += "uk.gov.hmrc.play.bootstrap.FrontendModule"
+play.modules.enabled += "uk.gov.hmrc.play.bootstrap.HttpClientModule"
+play.modules.enabled += "uk.gov.hmrc.play.audit.AuditModule"
+play.modules.enabled += "uk.gov.hmrc.play.bootstrap.graphite.GraphiteMetricsModule"
```
Comment (cyeniceri) [inline src/test/resources/application.conf:21]: is this file required ?

File: src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<configuration>
+    <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder class="ch.qos.logback.classic.encoder.PatternLayoutEncoder">
+            <pattern>%date{ISO8601} level=[%level] logger=[%logger] message=[%message]</pattern>
+        </encoder>
```
Comment (cyeniceri) [inline src/test/resources/logback-test.xml:5]: same as above comment, is it required ?

File: src/test/scala/connectors/ScaWrapperDataConnectorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      server.stubFor(
+        get(anyUrl()).willReturn(
+          aResponse()
+            .withStatus(OK)
+            .withBody("You haven't seen me, right!")
+            .withFixedDelay(2000)
+        )
+      )
+
+      val app = GuiceApplicationBuilder().configure(
```
Comment (cyeniceri) [inline src/test/scala/connectors/ScaWrapperDataConnectorSpec.scala:57]: If you pull main, you'll see the update from Adil's commit. Some internal configurations within this tests already done (like applicationBuilder, port config etc..)

**Review by cyeniceri — CHANGES_REQUESTED** [review]

File: src/test/resources/application.conf (hunk 1 above)
Comment (PerrettJ4) [reply src/test/resources/application.conf:21]: Yes otherwise there are 3 Errors and 1 failure printed to console

`[info] *** 3 SUITES ABORTED ***`
`[info] *** 1 TEST FAILED ***`

`[info] WrapperServiceSpec:
[info] service.WrapperServiceSpec *** ABORTED ***
[info]   com.google.inject.ProvisionException: Unable to provision, see the following errors:
[info] 
[info] 1) No implementation for com.kenshoo.play.metrics.MetricsFilter was bound.
[info]   while locating com.kenshoo.play.metrics.MetricsFilter
` etc....

File: src/test/scala/fixtures/Timeout.scala [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+package fixtures
+
+import scala.concurrent._
+import scala.concurrent.duration._
+import akka.actor.ActorSystem
+import play.api.Logging
+
+case object FutureEarlyTimeout extends RuntimeException
+
+trait Timeout extends Logging {
```
Comment (cyeniceri) [inline src/test/scala/fixtures/Timeout.scala:26]: this is used in BaseSpec, but BaseSpec is not used. what is this used for ?

Comment (cyeniceri) [reply src/test/scala/fixtures/Timeout.scala:26]: tests are still passing without it
